From 558d42b9bc8ea1e8ee2265d6473b485a93be3649 Mon Sep 17 00:00:00 2001
From: Iain Russell <Iain.Russell@ecmwf.int>
Date: Fri, 13 Nov 2020 18:01:52 +0000
Subject: [PATCH] Macro: improve performance of fieldset lookup function
 [METV-2836]

---
 src/Macro/grib.cc | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/metview/src/Macro/grib.cc b/metview/src/Macro/grib.cc
index 07733f1c..207dbab3 100644
--- a/metview/src/Macro/grib.cc
+++ b/metview/src/Macro/grib.cc
@@ -2652,6 +2652,9 @@ Value LookupFunction2::Execute(int, Value* arg)
     fieldset* z = copy_fieldset(v, v->count, false);
     int count   = w->count;
 
+    size_t prev = (size_t)-1;
+    field* k = NULL;
+
     for (i = 0; i < v->count; i++) {
         field* g = get_field(v, i, expand_mem);
         field* h = get_field(z, i, expand_mem);
@@ -2661,12 +2664,15 @@ Value LookupFunction2::Execute(int, Value* arg)
             if (n < 0 || n >= count) {
                 return Error("lookup: value out of range");
             }
-            field* k = get_field(w, n, expand_mem);
-            if (j >= k->value_count) {
-                return Error("lookup: fields mismatch");
+
+            if (n != prev) {
+                k = get_field(w, n, expand_mem);
+                prev = n;
+                if (j >= k->value_count) {
+                    return Error("lookup: fields mismatch");
+                }
             }
             h->values[j] = k->values[j];
-            release_field(k);
         }
 
         release_field(g);
@@ -2674,6 +2680,10 @@ Value LookupFunction2::Execute(int, Value* arg)
             save_fieldset(z);
     }
 
+    if (k) {
+        release_field(k);
+    }
+
     save_fieldset(z);
 
     return Value(z);
-- 
2.13.7
